src/runtime/propertyobject.cs

```diff
@@ -51,8 +56,8 @@ public static IntPtr tp_descr_get(IntPtr ds, IntPtr ob, IntPtr tp)
 
                 try
                 {
-                    result = self.info.GetValue(null, null);
-                    return Converter.ToPython(result, self.info.PropertyType);
+                    result = self.info.Value.GetValue(null, null);
+                    return Converter.ToPython(result, self.info.Value.PropertyType);
```

Comment:
I'm seeing this idiom a lot: at the top of the function, check if valid; at the end, get the Value without checking for exceptions.

I think that we should get the value immediately after checking if it's valid, so we don't somehow refactor and end up getting Value without checking for validity at all.